shoop/admin/modules/orders/views/edit.py

```diff
@@ -332,6 +317,92 @@ def handle_finalize(self, request):
+    order_type = "sales"
+
+    def get_context_data(self, **kwargs):
+        context = super(OrderEditView, self).get_context_data(**kwargs)
+        customer_id = self.request.GET.get("contact_id")
+        shipping_methods = ShippingMethod.objects.enabled()
+        payment_methods = PaymentMethod.objects.enabled()
+        context["config"].update({
+            "countries": [{"id": code, "name": name} for code, name in list(countries)],
+            "shippingMethods": [encode_method(sm) for sm in shipping_methods],
+            "paymentMethods": [encode_method(pm) for pm in payment_methods],
+            "customerData": self.get_customer_data(customer_id) if customer_id else None
+        })
+        return context
+
+    def get_customer_data(self, customer_id):
+        customer = Contact.objects.filter(pk=customer_id).first()
+        if not customer:
+            return JsonResponse(
+                {"success": False, "errorMessage": _("Contact %s does not exist.") % customer_id}, status=400
+            )
+        return {
+            "id": customer.id,
+            "name": customer.name,
+            "isCompany": bool(isinstance(customer, CompanyContact)),
+            "billingAddress": encode_address(customer.default_billing_address),
+            "shippingAddress": encode_address(customer.default_shipping_address)
+        }
+
+    def handle_customer_data(self, request):
+        customer_id = request.GET["id"]
+        return self.get_customer_data(customer_id)
+
+
+class PurchaseOrderEditView(BaseOrderEditView):
+    model = PurchaseOrder
+    order_type = "purchase"
+
+    def get_context_data(self, **kwargs):
+        context = super(PurchaseOrderEditView, self).get_context_data(**kwargs)
+        manufacturer = None
+        manufacturer_id = self.request.GET.get("manufacturer_id")
+        if manufacturer_id:
+            manufacturer = Manufacturer.objects.filter(pk=manufacturer_id).first()
+        context["config"].update({
+            "manufacturer": encode_manufacturer_data(manufacturer) if manufacturer else None
+        })
+        return context
+
+    def handle_manufacturer_data(self, request):
+        manufacturer_id = request.GET["id"]
+        manufacturer = Manufacturer.objects.filter(pk=manufacturer_id).first()
+        if not manufacturer:
+            return JsonResponse(
+                {"success": False, "errorMessage": _("Manufacturer %s does not exist.") % manufacturer_id}, status=400
+            )
+        return encode_manufacturer_data(manufacturer)
+
+    def handle_product_data(self, request):
+        product_data = super(PurchaseOrderEditView, self).handle_product_data(request)
+        product = Product.objects.filter(pk=product_data["id"]).first()
+        shop_id = request.GET["shop_id"]
+        shop = Shop.objects.get(pk=shop_id)
+        shop_product = product.get_shop_instance(shop)
+        supplier = shop_product.suppliers.first()  # TODO: Allow setting a supplier?
```

Comment:
I think you might want to make supplier selectable for merchant or make this first simple supplier.